garage/tf/policies/deterministic_mlp_policy_with_model.py

```diff
@@ -0,0 +1,100 @@
+"""DeterministicMLPPolict with model."""
+from akro.tf import Box
+import tensorflow as tf
+
+from garage.misc.overrides import overrides
+from garage.tf.models.discrete_mlp_model import DiscreteMLPModel
+from garage.tf.policies.base2 import Policy2
+
+
+class DeterministicMLPPolicyWithModel(Policy2):
```

Comment:
What is the difference of DeterministicMLPPolicyWithModel and ContinuousMLPPolicyWithModel? We should only keep one in the refactoring if they are the same. I also don’t think DeterministicPolicy should only work with Box space.

Reply:
ContinuousMLPPolicyWithModel uses ContinuousMLPModel which scales the output. I suggest we just keep one MLPModel which has default `output_scale` = 1.

I think discussion are needed for how precise we want to have the policy to be. Do we need to construct all the 4 possible policies? i.e. `Discrete/Continuous` * `Deterministic/Stochastic`.

Reply:
First, the `output_scale` is not important/necessary there because the action value will be scaled in the environment wrapper. 
Second, `ContinuousMLPPolicy` and `DeterministicMLPPolicy` are the same in concept. We have two of them because the `DeterministicMLPPolicy` was implemented long ago and never tested. The goal is to keep only one of them eventually. 
Third, we can follow our current design, based on the type of action space, the stochastic policy will model the corresponding type of distribution. And by default, the deterministic policy is for continuous space.

Reply:
Let's have only `DeterministicMLPPolicyWithModel` and remove `output_scale`?